kiwi/repository/zypper.py

```diff
@@ -263,6 +263,20 @@ def delete_all_repos(self):
         Path.wipe(self.shared_zypper_dir['reposd-dir'])
         Path.create(self.shared_zypper_dir['reposd-dir'])
 
+    def delete_repo_cache(self, name):
+        """
+        Delete zypper repository cache
+
+        The cache data for each repository is stored in a directory
+        of the same name as the repository name. The method deletes
+        this directory to cleanup the cache information
+
+        :param string name: repository name
+        """
+        Path.wipe(
+            os.sep.join([self.shared_zypper_dir['pkg-cache-dir'], name])
+        )
+
```

Comment:
@schaefi this is not enough for zypper, since this only cleans cached packages, but not cached repository metadata (/var/cache/kiwi/zypper/raw/<repo> and there are other directories inside `zypper` folder which could be deleted, even they do not contain any signature). With this code the repository is still trusted if it was already on the cache. The build will fail, but due to package singature check, not due to repository signature check. 

As a side note not related with this PR, I just noticed that if the repository signature fails KIWI does not fail even the repository is automatically rejected, it just continues without the untrusted repository, which is likely to fail due to some missing packages, but could not be the case (imagine update repo with a different key).